apps/astarte_pairing/lib/astarte_pairing/queries.ex

```diff
@@ -123,14 +119,10 @@ defmodule Astarte.Pairing.Queries do
   end
 
   def unregister_device(realm_name, device_id) do
-    with :ok <- verify_already_registered_device(realm_name, device_id),
-         :ok <- do_unregister_device(realm_name, device_id) do
+    with {:ok, device} <- fetch_device(realm_name, device_id),
+         {:ok, _device} <- do_unregister_device(realm_name, device) do
```

Comment:
Just a comment, I think this is ok, but it is causing a problem in the `pairing_api`, maybe you can provide a fix in this same PR